animation_nodes/base_types/tree_auto_execution.py

```diff
@@ -2,83 +2,104 @@
 from bpy.props import *
 from .. utils.blender_ui import getDpiFactor
 
-triggerTypeItems = [
-    ("MONITOR_PROPERTY", "Monitor Property", "", "", 0)]
-
 idTypeItems = [
     ("OBJECT", "Object", "", "OBJECT_DATA", 0),
-    ("SCENE", "Scene", "", "SCENE_DATA", 1)]
+    ("COLLECTION", "Collection", "", "GROUP", 1),
+    ("SCENE", "Scene", "", "SCENE_DATA", 2)]
 
 class AutoExecutionTrigger_MonitorProperty(bpy.types.PropertyGroup):
     bl_idname = "an_AutoExecutionTrigger_MonitorProperty"
 
+    def resetIDs(self, context):
+        self.object = self.collection = self.scene = None
+
     idType: EnumProperty(name = "ID Type", default = "OBJECT",
-        items = idTypeItems)
+        items = idTypeItems, update = resetIDs)
 
     object: PointerProperty(type = bpy.types.Object, name = "Object")
+    collection: PointerProperty(type = bpy.types.Collection, name = "Collection")
     scene: PointerProperty(type = bpy.types.Scene, name = "Scene")
-    dataPath: StringProperty(name = "Data Path", default = "")
+    dataPaths: StringProperty(name = "Data Paths", default = "",
+        description = "Comma separated paths of properties to monitor")
 
     lastState: StringProperty(default = "")
+    expanded: BoolProperty(default = True)
     enabled: BoolProperty(default = True)
     hasError: BoolProperty(default = False)
 
     def update(self):
         lastState = self.lastState
-        newState = self.getPropertyState()
+        newState = self.getPropertiesState()
         if newState == lastState:
             return False
         else:
             self.lastState = newState
             return self.enabled and lastState != ""
 
-    def getPropertyState(self):
-        prop = self.getProperty()
-        if prop is None: return ""
-        if hasattr(prop, "__iter__"):
-            return " ".join(str(part) for part in prop)
-        return str(prop)
+    def getPropertiesState(self):
+        props = self.getProperties()
+        if len(props) == 0: return ""
 
-    def getProperty(self):
-        self.hasError = False
-        object = self.getObject()
-        if object is None or self.dataPath is "":
-            return None
+        propsString = ""
+        for prop in props:
+            if hasattr(prop, "__iter__"):
+                propsString += "".join(str(part) for part in prop)
+            else: propsString += str(prop)
 
-        try: return object.path_resolve(self.dataPath)
+        return propsString
+
+    def getProperties(self):
+        self.hasError = False
+        IDBlocks = self.getIDBlocks()
+        if len(IDBlocks) == 0 or self.dataPaths.strip() is "": return []
+
+        try:
+            properties = []
+            for IDBlock in IDBlocks:
+                properties.extend(IDBlock.path_resolve(p.strip()) for p in self.dataPaths.split(","))
```

Comment:
Better make a utility getter that prepares a list of data paths.

Reply:
You mean something like this:
```python
    def getDataPaths(self):
        return [p.strip() for p in self.dataPaths.split(",")]
```

Reply:
yes